### PR TITLE
Revised README.md

### PR DESCRIPTION
**Pull request by jparise**
This is a full revision of the README.md that brings it up to date with
our current code. It also rephrases many of the example snippets in
terms of the new Calculator example project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,95 +53,117 @@ defmodule MyProject.Mixfile do
...
-creates a [Behaviour](http://elixir-lang.org/docs/stable/elixir/behaviours.html#content),
-complete with correct success typing, for this module. To implement this behaviour,
-use the `@behaviour` module attribute. The compiler will now inform you about
-any missed functions.
+## Servers
 
-Here is an implementation for the server defined above:
+In order to start a Thrift server, you will need to provide a callback module
+that implements the functions described by its service interface. Fortunately,
+a [behaviour][] module will be automatically generated for you, complete with
```
Comment (thecodeboss) [inline README.md]: I don't think you need the extra `[]` in your link here, at least according to [this markdown cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links).

File: README.md [hunk 2]
```diff
@@ -48,95 +53,117 @@ defmodule MyProject.Mixfile do
...
 
 ```elixir
-{:ok, server_pid} = Thrift.Test.UserService.Binary.Framed.Server.start_link(UserServiceHandler, 2345, [])
+iex> alias Calculator.Generated.Service.Binary.Framed.Server
+iex> {:ok, server} = Server.start_link(CalculatorHandler, 9090, [])
 ```
 
-...and your server is up and running. RPC calls to the server are delegated to UserServiceHandler.
+All RPC calls to the server will be delegated to the handler module. The server
+provides a [supervisor][] which can be added to your application's supervision
```
Comment (thecodeboss) [inline README.md]: same here

File: README.md [hunk 3]
```diff
@@ -48,95 +53,117 @@ defmodule MyProject.Mixfile do
...
-  def get_user_by_id(user_id) do
-    case Backend.find_user_by_id(user_id) do
-      {:ok, user} ->
-        user
-      {:error, _} ->
-        raise Thrift.Test.UserNotFound.exception message: "could not find user with id #{user_id}"
-    end
-  end
+defmodule CalculatorHandler do
+  @behaviour Calcualtor.Generated.Service.Handler
```
Comment (thecodeboss) [inline README.md]: s/Calcualtor/Calculator/

File: README.md [hunk 4]
```diff
@@ -48,95 +53,117 @@ defmodule MyProject.Mixfile do
...
 ```elixir
-{:ok, server_pid} = Thrift.Test.UserService.Binary.Framed.Server.start_link(UserServiceHandler, 2345, [])
+iex> alias Calculator.Generated.Service.Binary.Framed.Server
+iex> {:ok, server} = Server.start_link(CalculatorHandler, 9090, [])
 ```
 
-...and your server is up and running. RPC calls to the server are delegated to UserServiceHandler.
+All RPC calls to the server will be delegated to the handler module. The server
+provides a [supervisor][] which can be added to your application's supervision
+tree. It's important to start it as a `supervisor` and not as a `worker`.
```
Comment (thecodeboss) [inline README.md]: Maybe make this more explicit by saying:
``It's important to add it to your supervision tree with type `:supervisor` and not `:worker`.``

File: README.md [hunk 5]
```diff
@@ -154,56 +181,46 @@ files. It is built on a low-level Erlang lexer and parser:
...
+project.
 
 Presently, we implement:
 
 * Binary Protocol, Framed Client
 * Binary Protocol, Framed Server
 
-We are more than willing to take PRs to add more. We intend to also add TMux and Finagle servers
-in the future.
+We are more than willing to access contributions that add more!
```
Comment (thecodeboss) [inline README.md]: s/access/accept/

File: README.md [hunk 6]
```diff
@@ -48,95 +53,118 @@ defmodule MyProject.Mixfile do
...
+```elixir
+Thrift.Parser.parse("enum Colors { RED, GREEN, BLUE }")
+%Thrift.AST.Schema{constants: %{},
+ enums: %{Colors: %Thrift.AST.TEnum{name: :Colors,
+    values: [RED: 1, GREEN: 2, BLUE: 3]}}, exceptions: %{}, includes: [],
+ namespaces: %{}, services: %{}, structs: %{}, thrift_namespace: nil,
+ typedefs: %{}, unions: %{}}
+```
+
+There is also a low-level lexer (`:thrift_lexer`) and parser (`:thrift_parser`)
```
Comment (fishcakez) [inline README.md]: Do we want to document these? I dont know if we want to be forced to support the intermediate tokens?

Comment (jparise) [reply README.md]: They've been there for a long time, but I do think it makes sense to remove them.

**Pull request merged**